core/src/main/java/org/apache/brooklyn/core/typereg/RegisteredTypes.java

```diff
@@ -66,180 +94,131 @@ public static RegisteredType of(CatalogItem<?, ?> item) {
-        public Collection<OsgiBundleWithUrl> getLibraries() {
-            return bundles;
+    @Beta
+    public static RegisteredType addSuperType(RegisteredType type, @Nullable RegisteredType superType) {
+        if (superType!=null) {
+            if (isSubTypeOf(superType, type)) {
+                throw new IllegalStateException(superType+" declares "+type+" as a supertype; cannot set "+superType+" as a supertype of "+type);
+            }
+            ((BasicRegisteredType)type).superTypes.add(superType);
         }
+        return type;
+    }
 
-        @Override
-        public String getDisplayName() {
-            return displayName;
-        }
+    /** returns the implementation data for a spec if it is a string (e.g. plan yaml or java class name); else false */
+    @Beta
+    public static String getImplementationDataStringForSpec(RegisteredType item) {
+        if (item==null || item.getPlan()==null) return null;
+        Object data = item.getPlan().getPlanData();
+        if (!(data instanceof String)) return null;
+        return (String)data;
+    }
 
-        @Override
-        public String getDescription() {
-            return description;
-        }
+    /** returns an implementation of the spec class corresponding to the given target type;
+     * for use in {@link BrooklynTypePlanTransformer#create(RegisteredType, RegisteredTypeLoadingContext)} 
+     * implementations when dealing with a spec; returns null if none found
+     * @param mgmt */
+    @Beta
+    public static AbstractBrooklynObjectSpec<?,?> newSpecInstance(ManagementContext mgmt, Class<? extends BrooklynObject> targetType) throws Exception {
+        Class<? extends AbstractBrooklynObjectSpec<?, ?>> specType = RegisteredTypeLoadingContexts.lookupSpecTypeForTarget(targetType);
+        if (specType==null) return null;
+        Method createMethod = specType.getMethod("create", Class.class);
+        return (AbstractBrooklynObjectSpec<?, ?>) createMethod.invoke(null, targetType);
+    }
 
-        @Override
-        public String getIconUrl() {
-            return iconUrl;
-        }
-        
-        @Override
-        public boolean isDisabled() {
-            return disabled;
-        }
-        
-        @Override
-        public boolean isDeprecated() {
-            return deprecated;
-        }
-        
-        @Override
-        public Class<?> getJavaType() {
-            return javaType;
-        }
-        
-        @Override
-        public String toString() {
-            return JavaClassNames.simpleClassName(this)+"["+getId()+
-                (isDisabled() ? ";DISABLED" : "")+
-                (isDeprecated() ? ";deprecated" : "")+
-                "]";
-        }
+    /** Returns a wrapped map, if the object is YAML which parses as a map; 
+     * otherwise returns absent capable of throwing an error with more details */
+    @SuppressWarnings("unchecked")
+    public static Maybe<Map<Object,Object>> getAsYamlMap(Object planData) {
```

Comment:
Return type should be `Maybe<Map<?,?>>` (it's a read-only container), callers are not supposed to mutate it.